finder-app/writer.c

```diff
@@ -0,0 +1,57 @@
+// Tester code for assignment 2 point 3)
+// File:	writer.c
+// Author:	Mukta Darekar
+
+
+// Pre-processor directives
+#include <stdlib.h>
+#include <stdint.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <stdbool.h>
+#include <string.h>
+#include <ctype.h>
+#include <syslog.h>
+#include <unistd.h>
+
+#define TOTAL_ARG	3	//including ./writer + filename + string = 3 arguments
+
+
+int main(int argc, char *argv[])
+{
+
+	openlog(NULL, LOG_CONS, LOG_USER);
+	
+	// verify arguments
+	if(argc != 3)
+	{
+		syslog(LOG_ERR, "Invalid number of arguments, need Filepath>Filename and string to write\n");
+		 return 1;
+	}
+			
+	int fd = open(argv[1], O_CREAT | O_RDWR | O_TRUNC, 0664);
```

Comment:
@MuktaDarekar a suggestion, rather than referencing argv[1] everywhere, you can use a descriptive name like:
```
const char *filepath=argv[1];
```
and also decouple the usage from the specific argument.

Reply:
Done.